### PR TITLE
Remove VerificationMaterials (take 2)

### PR DESCRIPTION
**Pull request by woodruffw**
~~Very WIP, still thinking this through.~~

TODO:

- [x] Bring the coverage back up a bit.
- [x] This changeset currently reverts our protections against Rekor entry confusion; I need to reintroduce those.
- [x] CHANGELOG entries

### CODE REVIEW

**Comment by woodruffw** [comment]
To summarize, roughly:

* `VerificationMaterials` is now entirely gone: APIs that took it now take a `sigstore.verify.Bundle` instead, which is a wrapper around the `protobuf_specs`-defined `Bundle` type.
* We now require an inclusion proof, full stop. This removes a bunch of special-casing around "detached" materials and older bundle types, and removes the need to test those edge cases.
* Bundles are now *always* verified offline. 
* Detached materials are now *always* verified online. Passing `--offline` with detached inputs causes an error.

**Comment by woodruffw** [comment]
Conformance failures here are caused by a dep mismatch: https://github.com/sigstore/sigstore-conformance/pull/132

**Comment by woodruffw** [comment]
This is a pretty chunky PR, so apologies in advance 😅

**Review by jku — APPROVED** [review]
Looks good, I have no design complaints: This seems like a logical simplification.

The code comments are very minor so approving as is.

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +59,25 @@ All versions prior to 0.9.0 are untracked.
   an `IO[bytes]` for input. Other input types (such as `Hashed` and
   `Statement`) are unchanged ([#921](https://github.com/sigstore/sigstore-python/pull/921))
 
+* **BREAKING API CHANGE**: `Verifier.verify(...)` now takes a `Bundle`,
+  instead of a `VerificationMaterials` ([#937](https://github.com/sigstore/sigstore-python/pull/937))
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply CHANGELOG.md:63]: Yes, good call!

File: sigstore/verify/models.py [hunk 2]
```diff
@@ -286,14 +220,14 @@ def from_bundle(
...
-        tlog_entries = bundle.verification_material.tlog_entries
-        if len(tlog_entries) != 1:
-            raise InvalidMaterials(
-                f"expected exactly one log entry, got {len(tlog_entries)}"
-            )
-        tlog_entry = tlog_entries[0]
+        # Extract the log entry. For the time being, we expect
+        # bundles to only contain a single log entry.
+        try:
+            tlog_entry = self._inner.verification_material.tlog_entries[0]
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply sigstore/verify/models.py:228]: Ah, good catch -- this was unintended.

(Given that this is an unauthenticated envelope I see no difference to the attacker between "exactly one valid entry" and "N entries, of which the first is valid," but there's also no reason to silently change the behavior here 🙂)

**Review by jku — APPROVED** [review]
